File: wordpress-plugin/includes/Domain/DraftResult.php
<?php

declare(strict_types=1);

namespace MaterialCapture\Domain;

use DateTimeImmutable;

/**
 * The outcome of a successful draft creation, shaped to match the `201` response body
 * documented in docs/api-spec.md.
 */
final class DraftResult
{
    public function __construct(
        public readonly int $postId,
        public readonly string $status,
        public readonly string $title,
        public readonly ?string $editUrl,
        public readonly ?string $previewUrl,
        public readonly string $category,
        public readonly DateTimeImmutable $createdAt,
    ) {
    }
}

File: android/core/src/main/kotlin/io/github/dopodomani/wpsharetodraft/domain/DraftResult.kt
package io.github.dopodomani.wpsharetodraft.domain

import java.time.Instant

/** The outcome of a successful [Destination.send], mirroring the `201` body in docs/api-spec.md. */
data class DraftResult(
    val postId: Long,
    val status: String,
    val title: String,
    val editUrl: String?,
    val previewUrl: String?,
    val category: String,
    val createdAt: Instant,
)
